lib/manageiq/cross_repo/runner.rb

```diff
@@ -35,14 +36,25 @@ def run
 
     def run_tests
       with_test_env do
-        system!({"TRAVIS_BUILD_DIR" => test_repo.path.to_s}, "bash", "tools/ci/before_install.sh") if ENV["CI"] && File.exist?("tools/ci/before_install.sh")
-        system!(env_vars, "bin/setup")
-        system!(script_cmd)
+        require "yaml"
+        travis_yml = YAML.load_file(".travis.yml")
+
+        Array(travis_yml["before_install"]).each { |before_install| system!(env_vars, before_install) }
+        Array(travis_yml["install"]).each { |install| system!(env_vars, install) }
+        Array(travis_yml["before_script"]).each { |before_script| system!(env_vars, before_script) }
+
+        if script_cmd
+          system!(env_vars, script_cmd)
+        else
+          Array(travis_yml["script"]).each { |script| system!(env_vars, script) }
+        end
+
+        Array(travis_yml["after_script"]).each { |after_script| system!(env_vars, after_script) }
       end
     end
 
     def env_vars
-      {"MANAGEIQ_REPO" => core_repo.path.to_s}
+      {"MANAGEIQ_REPO" => core_repo.path.to_s, "TRAVIS_BUILD_DIR" => test_repo.path.to_s, "TEST_SUITE" => test_suite}
```

Comment:
Does the test_suite have to be defaulted to `spec` or `spec:vmdb` for core?

Reply:
Yes, I wonder if that is something we might have to fix in core though since a TEST_SUITE of `"vmdb"` won't work for most plugins...

Reply:
If core is inconsistent, which I believe it is, then I'm good with changing things for consistency.

Reply:
https://github.com/ManageIQ/manageiq/pull/20984 allows us to call `rake test:setup` and `rake test` and pick up on `ENV["TEST_SUITE"]` while still defaulting to `"vmdb"`